File: etablissement/src/main/java/com/projetfinal/etablissement/entity/Adresse.java
package com.projetfinal.etablissement.entity;

import javax.persistence.Embeddable;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotEmpty;

import com.fasterxml.jackson.annotation.JsonView;

@Embeddable
public class Adresse {

	@NotEmpty
	@JsonView(Vue.Common.class)
	private String nomRue;
	@NotEmpty
	@JsonView(Vue.Common.class)
	private Integer numRue;
	@NotEmpty
	@JsonView(Vue.Common.class)
	private String codePostal;
	@NotEmpty
	@JsonView(Vue.Common.class)
	private String ville;
	
	
	public Adresse() {
	}

	public Adresse(@NotEmpty String nomRue, @NotEmpty Integer numRue, @NotEmpty String codePostal,
			@NotEmpty String ville) {
		this.nomRue = nomRue;
		this.numRue = numRue;
		this.codePostal = codePostal;
		this.ville = ville;
	}

	public String getNomRue() {
		return nomRue;
	}

	public void setNomRue(String nomRue) {
		this.nomRue = nomRue;
	}

	public Integer getNumRue() {
		return numRue;
	}

	public void setNumRue(Integer numRue) {
		this.numRue = numRue;
	}

	public String getCodePostal() {
		return codePostal;
	}

	public void setCodePostal(String codePostal) {
		this.codePostal = codePostal;
	}

	public String getVille() {
		return ville;
	}

	public void setVille(String ville) {
		this.ville = ville;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((codePostal == null) ? 0 : codePostal.hashCode());
		result = prime * result + ((nomRue == null) ? 0 : nomRue.hashCode());
		result = prime * result + ((numRue == null) ? 0 : numRue.hashCode());
		result = prime * result + ((ville == null) ? 0 : ville.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Adresse other = (Adresse) obj;
		if (codePostal == null) {
			if (other.codePostal != null)
				return false;
		} else if (!codePostal.equals(other.codePostal))
			return false;
		if (nomRue == null) {
			if (other.nomRue != null)
				return false;
		} else if (!nomRue.equals(other.nomRue))
			return false;
		if (numRue == null) {
			if (other.numRue != null)
				return false;
		} else if (!numRue.equals(other.numRue))
			return false;
		if (ville == null) {
			if (other.ville != null)
				return false;
		} else if (!ville.equals(other.ville))
			return false;
		return true;
	}

	@Override
	public String toString() {
		return "Adresse [nomRue=" + nomRue + ", numRue=" + numRue + ", codePostal=" + codePostal + ", ville=" + ville
				+ "]";
	}
	
	
}

File: etablissementAngular/src/app/model/adresse.ts
export class Adresse {
    private _id!: number;
    private _numRue: number;
    private _nomRue: string;
    private _codePostal: string;
    private _ville: string;


	constructor(id?: number, numRue: number = 0, nomRue: string='', codePostal:string='', ville: string='') {
		this._numRue = numRue;
		this._nomRue = nomRue;
        this._codePostal = codePostal;
		this._ville = ville;
	}


    /**
     * Getter codePostal
     * @return {string}
     */
	public get codePostal(): string {
		return this._codePostal;
	}

    /**
     * Setter codePostal
     * @param {string} value
     */
	public set codePostal(value: string) {
		this._codePostal = value;
	}

    /**
     * Getter id
     * @return {number}
     */
	public get id(): number {
		return this._id;
	}

    /**
     * Getter numRue
     * @return {number}
     */
	public get numRue(): number {
		return this._numRue;
	}

    /**
     * Getter nomRue
     * @return {string}
     */
	public get nomRue(): string {
		return this._nomRue;
	}

    /**
     * Getter ville
     * @return {string}
     */
	public get ville(): string {
		return this._ville;
	}

    /**
     * Setter id
     * @param {number} value
     */
	public set id(value: number) {
		this._id = value;
	}

    /**
     * Setter numRue
     * @param {number} value
     */
	public set numRue(value: number) {
		this._numRue = value;
	}

    /**
     * Setter nomRue
     * @param {string} value
     */
	public set nomRue(value: string) {
		this._nomRue = value;
	}

    /**
     * Setter ville
     * @param {string} value
     */
	public set ville(value: string) {
		this._ville = value;
	}


}
